### DISCUSSION TITLE
Selected text range through Accessibility APIs (macOS)

### DISCUSSION THREAD

**Discussion by joshmedeski** [neovim/neovim#28568] [Ideas]
Neovim currently has a problem with selected text in macOS. It doesn't correctly expose the selected text range through Accessibility APIs (it correctly exposes the cursor position, but not the selected text range). This forbids all the workflows depending on text selection (like Raycast and screen readers).

I realize there is probably some integration to be done with the terminal emulator. But when it comes to Neovim has any work been done to try and integrate with macOS accessibility features like selected text?

**Comment by clason** [comment]
This _needs_ to be communicated through the terminal, so the first step would be to find out if 1. there is a terminal that uses these features and 2. there is an OSC to communicate text selection.

If either of these questions has a negative answer, your best bet is to look at macOS GUIs (like VimR).
